geotrek/api/v2/filters.py

```diff
@@ -753,17 +753,24 @@ def filter_queryset_related_objects_published_not_deleted(self, queryset, reques
         qs = queryset
         # Exclude if no related objects exist
         qs = qs.exclude(**{'{}'.format(prefix): None})
-        language = request.GET.get('language', 'all')
         q = Q()
-        if language == 'all':
-            # no language specified. Check for all.
-            for lang in settings.MODELTRANSLATION_LANGUAGES:
-                related_field_name = '{}__published_{}'.format(prefix, lang)
-                q |= Q(**{related_field_name: True})
-        else:
-            # one language is specified
-            related_field_name = '{}__published_{}'.format(prefix, language)
+        # check if the model of the queryset published field is translated
+        associated_published_fields = [f.name for f in qs.model._meta.get_field(prefix).remote_field.model._meta.get_fields() if f.name.startswith('published')]
```

Comment:
Maybe this line could be narrowed if you set a variable with the huge queryset
```suggestion
        associated_fields = qs.model._meta.get_field(prefix).remote_field.model._meta.get_fields()
        associated_published_fields = [f.name for f in translated_fields if f.name.startswith('published')]
```